lib/web/service-worker/interceptor.js

```diff
@@ -0,0 +1,13 @@
+import { BatchInterceptor } from '@mswjs/interceptors'
+import { ClientRequestInterceptor } from '@mswjs/interceptors/ClientRequest'
+import { FetchInterceptor } from '@mswjs/interceptors/fetch'
+import { XMLHttpRequestInterceptor } from '@mswjs/interceptors/XMLHttpRequest'
+
+export const interceptor = new BatchInterceptor({
```

Comment:
## Question

Which interception algorithm do you see more fitting?

I was considering between the `MockAgent` of Undici and Interceptors from MSW. There are reasons to use either. 

- `MockAgent` is in-line with Undici but covers _only_ Undici's implementation of `fetch`. I'd expect the Service Worker API to intercept _any_ HTTP requests, just like it does in the browser.
- Interceptors intercepts any HTTP requests but is a third-party library. If you are willing to add it as a dependency, I have nothing against using it. Interceptors is also slowly becoming the backbone for request interceptions for other libraries too, with Nock [slowly migrating](https://github.com/nock/nock/pull/2517) to use it instead. Using it in Undici may solidify it as a well-established request interception API in Node.js.